### ISSUE TITLE
amp.Integer is either broken or incorrectly documented

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2650]
|[<img alt="bitglue's avatar" src="https://avatars.githubusercontent.com/u/1232472?s=50" width="50" height="50">](https://github.com/bitglue)| @bitglue reported|
|-|-|
|Trac ID|trac#2650|
|Type|defect|
|Created|2007-05-15 19:20:32Z|

"Convert to and from 'int'."

But actually, it converts to and from 'int' and maybe 'long'. If it is intended to not be restricted to probably 32 bit numbers like 'int' (A Good Idea, i think) then this should be explicitly documented, and the documentation should not say 'int'.

Attachments:

* [2650.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/50b/50bda5d853560131e15ef9db3ec101d60ee3f26a/f233f74350d6d23c92d82374fa0fa24ca86820c3.diff) (1741 bytes) - added by zaheerm on 2010-03-21 11:40:30Z - Fixes docstring and adds unit test

<details><summary>Searchable metadata</summary>

```
trac-id__2650 2650
type__defect defect
reporter__indigo indigo
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__amp__documentation amp, documentation
time__1179256832000000 1179256832000000
changetime__1269192120000000 1269192120000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

The difference between int and long tends to disappear in Python. I even think long will disappear in Py3K. Some _maybe_ not adding long is a good idea. There is definitely something to be done about the documentation though.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @bitglue|
|-|-|

Yeah, the documentation could stand to be improved in more ways than just this.  (The implementation is probably a bit too lenient as well; simply ```int()```ing the argument allows for passing strings through and some other probably-not-so-good things.)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

Just to be explicit, it is indeed _not_ the intent that Integer be limited to 32 bit values.

**Comment by twisted-trac** [comment]
|[<img alt="zaheerm's avatar" src="https://avatars.githubusercontent.com/u/68943?s=50" width="50" height="50">](https://github.com/zaheerm)|@zaheerm removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Great.  The copyright dates on the modified files should be updated, and there needs to be a news file, but I'll take care of that and apply the patch.  Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [28760]) Apply amp.Integer improvement patch

Author: thomasvs, zaheerm
Reviewer: exarkun
Fixes: [#2650](https://github.com/twisted/twisted/issues/2650)

Clarify the API documentation for twisted.protocols.amp.Integer by mentioning
that integers of any size are supported and adding an example.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
